test/driver_test.rb

```diff
@@ -128,9 +133,154 @@
+      #FEE = 1.65
+      @driver = RideShare::Driver.new(
+          id: 54,
+          name: "Rogers Bartell IV",
+          vin: "1C9EVBRM0YBC564DZ"
+      )
+      @trip_1 = RideShare::Trip.new(
+          id: 8,
+          driver: @driver,
+          passenger_id: 3,
+          start_time: Time.new(2016, 8, 8),
+          end_time: Time.new(2016, 8, 8),
+          rating: 5,
+          cost: 15.0,
+      )
+      @trip_2 = RideShare::Trip.new(
+          id: 9,
+          driver: @driver,
+          passenger_id: 4,
+          start_time: Time.new(2016, 9, 8),
+          end_time: Time.new(2016, 9, 9),
+          rating: 5,
+          cost: 10,
+          )
+      @passenger = RideShare::Passenger.new(
+          id: 9,
+          name: "Merl Glover III",
+          phone_number: "1-602-620-2330 x3723",
+          trips: []
+      )
+      #@driver.add_trip(trip_1)
+    end
     # You add tests for the total_revenue method
+    it "returns an float " do
+      @driver.add_trip(@trip_1)
+
+      #assert
+      expect(@driver.total_revenue).must_be_kind_of Float
+    end
+
+    it "correctly get the total for a single trip" do
+      driver = RideShare::Driver.new(
+          id: 54,
+          name: "Rogers Bartell IV",
+          vin: "1C9EVBRM0YBC564DZ"
+      )
+      trip = RideShare::Trip.new(
+          id: 8,
+          driver: @driver,
+          passenger_id: 3,
+          start_time: Time.new(2016, 8, 8),
+          end_time: Time.new(2016, 8, 8),
+          rating: 5,
+          cost: 1,
+          )
+      driver.add_trip(trip)
+
+      expect(driver.total_revenue).must_equal 0
+
+
+    end
+
+    it "correctly calculates the total revenue of two trips" do
+      #arrange and act
+      @driver.add_trip(@trip_1)
+      @driver.add_trip(@trip_2)
+
+      #assert
+      expect(@driver.total_revenue).must_be_close_to 17.36, 0.01
+
+    end
+
+    it "returns zero if no driven trips" do
+      # assert
+      expect(@driver.total_revenue).must_equal 0
+
+    end
+
+    it "ignores in-progress trips" do
+
+      new_trip = RideShare::Trip.new(
+          # what should the id of this trip be?
+          id: 25,
+          passenger: @passenger,
+          passenger_id: @passenger.id,
+          start_time: Time.now,
+          end_time: nil,
+          cost: nil,
+          rating: nil,
+          driver_id: 4,
+          )
+
+      @driver.add_trip(@trip_1)
+      @driver.add_trip(new_trip)
+
+      expect(@driver.total_revenue).must_be_close_to 10.68, 0.1
+
+    end
+
   end
+  # tests for wave 3: change_status
+  describe "change status" do
+    before do
+      @driver = RideShare::Driver.new(
+          id: 60,
+          name: "Rogers Bartell IV",
+          vin: "1C9EVBRM0YBC564DZ",
+          status: :AVAILABLE
+      )
+    end
+
+    it "change the status of driver" do
+      #act and assert
+      expect(@driver.status).must_equal :AVAILABLE
+
+      expect(@driver.change_status).must_equal :UNAVAILABLE
+
+    end
+  end
```

Comment:
I think this `change_status` method should be a private method.   Unless you want clients to directly change the status, make it private.  Then you just test the public methods.